.pre-commit-config.yaml

```diff
@@ -7,19 +7,21 @@ exclude: |
   )$
 
 repos:
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        name: isort (python)
+        args: ["--settings-path", "python-package/pyproject.toml"]
```

Comment:
Proposing moving `isort` up above `ruff` because they were fighting each other. I think it makes sense to have `isort` go first since it might do things like remove duplicate imports or re-group them, and then `ruff` after which will simply reformat them.

Reply:
relevant conversation: https://github.com/astral-sh/ruff/issues/8926#issuecomment-1834048218

Reply:
ah nevermind this one, looks like this was a result of me adding `profile = "black"` to the `isort` config while testing. Reverted that in https://github.com/microsoft/LightGBM/pull/6317/commits/007ed3400f97906144d4ea50b4a4ee52aeef489f and `python-package/` changes dropped out of the diff.